File: libs/language/src/lib/utils/limits/NoLimits.ts
import { Limits } from './Limits';

/**
 * Implementation of Limits that imposes no limits on node extension
 */
export class NoLimits extends Limits {
  /**
   * Determines if a path segment should be extended - always returns true
   * @returns Always true
   */
  public shouldExtendPathSegment() {
    return true;
  }

  /**
   * Determines if a path segment should be merged - always returns true
   * @returns Always true
   */
  public shouldMergePathSegment() {
    return true;
  }

  /**
   * Enters a path segment - no-op
   */
  public override enterPathSegment() {
    // No-op
  }

  /**
   * Exits a path segment - no-op
   */
  public exitPathSegment() {
    // No-op
  }
}

File: references/java-sdk/src/main/java/blue/language/utils/limits/NoLimits.java
package blue.language.utils.limits;

import blue.language.model.Node;

class NoLimits implements Limits {

    @Override
    public boolean shouldExtendPathSegment(String pathSegment, Node currentNode) {
        return true;
    }

    @Override
    public boolean shouldMergePathSegment(String pathSegment, Node currentNode) {
        return true;
    }

    @Override
    public void enterPathSegment(String pathSegment, Node node) {
    }

    @Override
    public void exitPathSegment() {
    }
}